packages/start-slicemachine/src/legacyMigrations/migrateAssets.ts

```diff
@@ -0,0 +1,174 @@
+		}
+	});
+};
+
+const ensureOrGenerateMockFile = (
+	targetPathToMocks: string,
+	deprecatedPathToMocks: string,
+	validator: (str: string) => boolean,
+	generate: () => Record<string, unknown>,
+) => {
+	try {
+		const rawMockContent = (() => {
+			if (fsSync.existsSync(targetPathToMocks)) {
+				return fsSync.readFileSync(targetPathToMocks, "utf-8");
+			}
+			if (fsSync.existsSync(deprecatedPathToMocks)) {
+				return fsSync.readFileSync(deprecatedPathToMocks, "utf-8");
+			}
+
+			return null;
+		})();
+
+		const regeneratedMocksString = JSON.stringify(generate(), null, 2);
+		const isValidMock = (() => {
+			try {
+				return rawMockContent !== null && validator(rawMockContent);
+			} catch {
+				return false;
+			}
+		})();
+
+		if (!isValidMock) {
+			fsSync.writeFileSync(targetPathToMocks, regeneratedMocksString);
+		} else if (rawMockContent) {
+			fsSync.writeFileSync(targetPathToMocks, rawMockContent);
+		} else {
+			fsSync.writeFileSync(targetPathToMocks, regeneratedMocksString);
+		}
+	} catch (error) {}
+};
+
+// TODO: MIGRATION - Move this to the Migration Manager
+export const migrateAssets = async (
+	manager: SliceMachineManager,
+): Promise<void> => {
+	try {
+		const allSlices = await manager.slices.readAllSlices();
+		const sharedSlices = allSlices.models.reduce(
+			(o, slice) => ({ ...o, [slice.model.id]: slice.model }),
+			{},
+		);
+
+		allSlices.models.forEach((c) => {
+			const targetPathToSliceFolder = path.join(
+				manager.cwd,
+				c.libraryID,
+				c.model.name,
+			);
+			const targetPathToMocks = path.join(
+				targetPathToSliceFolder,
+				MOCKS_FILE_NAME,
+			);
+			const deprecatedPathToSliceAssets = path.join(
+				createPathToDeprecatedLibrary(manager.cwd),
+				"assets",
+				c.libraryID,
+				c.model.name,
+			);
+			const deprecatedPathToMocks = path.join(
+				deprecatedPathToSliceAssets,
+				MOCKS_FILE_NAME,
+			);
+
+			ensureOrGenerateMockFile(
+				targetPathToMocks,
+				deprecatedPathToMocks,
+				(str: string) =>
+					SharedSliceContent.decode(JSON.parse(str))._tag === "Right",
+				() => SharedSliceMock.generate(c.model),
+			);
+
+			const variationsIDs = c.model.variations.map((v) => v.id);
+			ensureOrGenerateSliceScreenshot(
+				variationsIDs,
+				targetPathToSliceFolder,
+				deprecatedPathToSliceAssets,
+			);
+			safeUnlink(deprecatedPathToMocks, "file");
+		});
+
+		const allCustomTypes = await manager.customTypes.readAllCustomTypes();
+		allCustomTypes.models.forEach((c) => {
+			const targetPathToMocks = path.join(
+				manager.cwd,
+				"customtypes",
+				c.model.id,
+				MOCKS_FILE_NAME,
+			);
+			const deprecatedPathToMocks = path.join(
+				createPathToCustomTypesAssets(manager.cwd),
+				c.model.id,
+				MOCKS_FILE_NAME,
+			);
+			ensureOrGenerateMockFile(
+				targetPathToMocks,
+				deprecatedPathToMocks,
+				(str: string) => Document.decode(JSON.parse(str))._tag === "Right",
+				() => DocumentMock.generate(c.model, sharedSlices),
+			);
+		});
+	} catch (error) {
```

Comment:
Is there anything to do here? Do we want to see the error in Sentry?

Reply:
Added in: [39e87f0](https://github.com/prismicio/slice-machine/pull/862/commits/39e87f06999e1fbc1cec298d8db2fe8eec2161c0), let me know!